test/test_propagate_filter/conftest.py

```diff
@@ -1,13 +1,22 @@
 from typing import Tuple
 
 import numpy as np
-
-# noinspection PyProtectedMember
+import pytest
+from numpy.random import default_rng
 from PyDynamic import shift_uncertainty
 
 
-def random_array(length):
-    return np.random.randn(length)
+@pytest.fixture
+def rng():
+    return default_rng()
+
+
+@pytest.fixture
+def random_standard_normal_array(rng):
+    def _generate_random_array(length):
+        return rng.standard_normal(length)
+
+    return _generate_random_array
```

Comment:
I would like to understand this better. In particular, what is the difference to 

`def random_standard_normal_array(rng=default_rng):` ?

Reply:
The design choice was kindof an intermediate result, where the `rng` fixture in fact was used outside of the `random_standard_normal_array` fixture. I included the randum number generator creation into the `random_standard_normal_array` fixture now in b03bd9f344dd66eb4455d865a7f534c3e44447fc.